include/boost/geometry/algorithms/detail/overlay/enrich_intersection_points.hpp

```diff
@@ -289,92 +266,69 @@ struct enriched_map_default_include_policy
     }
 };
 
+// Add all (non discarded) operations on this ring
+// Blocked operations or uu on clusters (for intersection)
+// should be included, to block potential paths in clusters
 template <typename Turns, typename MappedVector, typename IncludePolicy>
 inline void create_map(Turns const& turns, MappedVector& mapped_vector,
                        IncludePolicy const& include_policy)
 {
-    typedef typename boost::range_value<Turns>::type turn_type;
-    typedef typename turn_type::container_type container_type;
-    typedef typename MappedVector::mapped_type mapped_type;
-    typedef typename boost::range_value<mapped_type>::type indexed_type;
-
-    std::size_t index = 0;
-    for (typename boost::range_iterator<Turns const>::type
-            it = boost::begin(turns);
-         it != boost::end(turns);
-         ++it, ++index)
+    for_each_with_index(turns, [&](std::size_t index, auto const& turn)
```

Comment:
This makes the code a lot shorter and more readable.
Better view it side-by-side.